main.py

```diff
@@ -178,6 +178,33 @@ def createLogFile(formatter,logger):
     now = datetime.datetime.now()
     stat_file.write(str(now.year)+', '+str(now.month)+', '+str(now.day)+', '+str(now.hour)+', '+str(str(ctx.channel.id))+", "+str(str(ctx.channel))+", "+str(author)+", "+str(ctx.command)+", "+str(argument)+", "+str(ctx.invoked_with)+", "+str(ctx.invoked_subcommand)+"\n")
 
+@bot.listen()
+async def on_member_join(member):
+
+    if member is not None:
+        from helper_files.embed import embed as imported_embed
+
+        output="Hi, welcome to the SFU CSSS Discord Server's"+str(os.environ['BRANCH'])+" branch.\n"
+        output+="\tWe are a group of students who live to talk about classes and nerdy stuff.\n"
+        output+="\tIf you need help, please ping any of our Execs, Execs at large, or First Year Reps.\n"
+        output+="\n"
+        output+="\tOur general channels include some of the following:\n"
+        output+="\t#off-topic, where we discuss damn near anything.\n"
+        output+="\t#first-years, for students who are starting, or about to start their first year.\n"
+        output+="\t#discussion, for serious non-academic discussion. (Politics et al.)\n"
+        output+="\t#sfu-discussions, for all SFU related discussion.\n"
+        output+="\t#projects_and_dev, for non-academic tech/dev/project discussion.\n"
+        output+="\t#bot_commands_and_misc, for command testing to reduce spam on other channels.\n"
+        output+="\n"
+        output+="\n"
+        output+="\tWe also have a smattering of course specific Academic channels.\n"
+        output+="\tYou can give yourself a class role by running <.iam cmpt320> or create a new class by <.newclass cmpt316>\n"
+        output+="\tPlease keep Academic Honesty in mind when discussing course material here.\n"
+
+        eObj = imported_embed(title="Welcome to the SFU CSSS's "+str(os.environ['BRANCH'])+" Discord Channel", author=settings.BOT_NAME, avatar=settings.BOT_AVATAR, description=output)
```

Comment:
Server not Channel

Reply:
also theres only one CSSS server, so i dont see why the need to specify branch

Reply:
@TitanVJ I was debugging